src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs

```diff
@@ -1149,5 +1149,114 @@ public void DataClassAndStruct()
                 Diagnostic(ErrorCode.ERR_LocalDuplicate, "Y").WithArguments("Y").WithLocation(5, 27)
             );
         }
+
+        [Fact]
+        public void RecordInheritance()
+        {
+            var src = @"
+class A { }
+record B : A { }
+record C : B { }
+class D : C { }
+interface E : C { }
+struct F : C { }
+enum G : C { }";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (3,12): error CS8864: Records may only inherit from object or another record
+                // record B : A { }
+                Diagnostic(ErrorCode.ERR_BadRecordBase, "A").WithLocation(3, 12),
+                // (5,11): error CS8865: Only records may inherit from records.
+                // class D : C { }
+                Diagnostic(ErrorCode.ERR_BadInheritanceFromRecord, "C").WithLocation(5, 11),
+                // (6,15): error CS0527: Type 'C' in interface list is not an interface
+                // interface E : C { }
+                Diagnostic(ErrorCode.ERR_NonInterfaceInInterfaceList, "C").WithArguments("C").WithLocation(6, 15),
+                // (7,12): error CS0527: Type 'C' in interface list is not an interface
+                // struct F : C { }
+                Diagnostic(ErrorCode.ERR_NonInterfaceInInterfaceList, "C").WithArguments("C").WithLocation(7, 12),
+                // (8,10): error CS1008: Type byte, sbyte, short, ushort, int, uint, long, or ulong expected
+                // enum G : C
+                Diagnostic(ErrorCode.ERR_IntegralTypeExpected, "C").WithLocation(8, 10)
+            );
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void RecordInheritance2(bool emitReference)
+        {
+            var src = @"
+public class A { }
+public record B { }
+public record C : B { }";
+            var comp = CreateCompilation(src);
+
+            var src2 = @"
+record D : C { }
+record E : A { }
+interface F : C { }
+struct G : C { }
+enum H : C { }
+";
+
+            var comp2 = CreateCompilation(src2,
+                parseOptions: TestOptions.RegularPreview,
+                references: new[] {
+                emitReference ? comp.EmitToImageReference() : comp.ToMetadataReference()
+            });
+
+            comp2.VerifyDiagnostics(
+                // (3,12): error CS8864: Records may only inherit from object or another record
+                // record E : A { }
+                Diagnostic(ErrorCode.ERR_BadRecordBase, "A").WithLocation(3, 12),
+                // (4,15): error CS0527: Type 'C' in interface list is not an interface
+                // interface E : C { }
+                Diagnostic(ErrorCode.ERR_NonInterfaceInInterfaceList, "C").WithArguments("C").WithLocation(4, 15),
+                // (5,12): error CS0527: Type 'C' in interface list is not an interface
+                // struct F : C { }
```

Comment:
nit: these diagnostic comments seem out of date, which makes interpreting the test a bit confusing.

Reply:
Which one are you thinking of?

```
// (5,12): error CS0527: Type 'C' in interface list is not an interface
// struct F : C { }
```

this looks right, structs can only have interfaces in the base list.

Reply:
for example, `E` is a record type in the source, but the diagnostic comments make mention of an `interface E`.